File: app/Models/Enrollment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Enrollment extends Model
{
    protected $fillable = [
        'course_id',
        'student_id',
        'token_id',
        'status',
        'registered_at',
        'cancelled_at',
    ];

    protected $casts = [
        'registered_at' => 'datetime',
        'cancelled_at' => 'datetime',
    ];

    public function course()
    {
        return $this->belongsTo(Course::class);
    }

    public function student()
    {
        return $this->belongsTo(User::class, 'student_id');
    }

    public function token()
    {
        return $this->belongsTo(Token::class);
    }
}

File: applicationKT/app/src/main/java/com/englishapp/data/model/Enrollment.kt
package com.englishapp.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Enrollment(
    @Json(name = "id") val id: Long,
    @Json(name = "course_id") val courseId: Long,
    @Json(name = "student_id") val studentId: Long,
    @Json(name = "token_id") val tokenId: Long? = null,
    @Json(name = "status") val status: String = "registered",
    @Json(name = "registered_at") val registeredAt: String? = null,
    @Json(name = "cancelled_at") val cancelledAt: String? = null,
    @Json(name = "course") val course: Course? = null,
    @Json(name = "student") val student: User? = null,
    @Json(name = "created_at") val createdAt: String? = null,
    @Json(name = "updated_at") val updatedAt: String? = null
)
